File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/configuration/rabbitmq/docs/consumer/custom/annotation/ProductListener.kt
package io.micronaut.configuration.rabbitmq.docs.consumer.custom.annotation

// tag::imports[]
import io.micronaut.configuration.rabbitmq.annotation.Queue
import io.micronaut.configuration.rabbitmq.annotation.RabbitListener
import io.micronaut.context.annotation.Requires

import java.util.Collections
import java.util.HashSet
// end::imports[]

@Requires(property = "spec.name", value = "DeliveryTagSpec")
// tag::clazz[]
@RabbitListener
class ProductListener {

    val messages: MutableSet<Long> = Collections.synchronizedSet(HashSet())

    @Queue("product")
    fun receive(data: ByteArray, @DeliveryTag tag: Long) { // <1>
        messages.add(tag)
    }
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/consumer/custom/annotation/ProductListener.py
from typing import Annotated

from micronaut.context.annotation import Requires
# tag::imports[]
from java.lang import Long
from micronaut.rabbitmq.annotation import Queue, RabbitListener

from .DeliveryTag import DeliveryTag
# end::imports[]


@Requires(property="spec.name", value="DeliveryTagSpec")
# tag::clazz[]
@RabbitListener
class ProductListener:

    def __init__(self) -> None:
        self.messages: set[int] = set()

    @Queue("product")
    def receive(self, data: bytes, tag: Annotated[Long, DeliveryTag]) -> None:  # <1>
        self.messages.add(tag)
# end::clazz[]
